app/routes/__test__/action.send-rating.test.ts

```diff
@@ -31,14 +31,18 @@ describe("/action/send-rating route", () => {
   });
 
   it("returns redirect without JS", async () => {
-    const ratingPath = "/relative/path#user-feedback-banner";
+    const ratingPath = "/relative/path?wasHelpful=yes#user-feedback-banner";
     const request = new Request(
       `http://localhost:3000/action/send-rating?url=${ratingPath}&js=false`,
       options,
     );
     const response = await action({ request, params: {}, context: {} });
     expect(response.status).toEqual(302);
-    expect(response.headers.get("location")).toEqual(ratingPath);
+    const location =
+      new URL(response.headers.get("location") ?? "").pathname +
+      new URL(response.headers.get("location") ?? "").search +
+      new URL(response.headers.get("location") ?? "").hash;
```

Comment:
instead of building 3 URL objects, could we destructure one, like so?:
```suggestion
    const {pathname, search, hash} = new URL(response.headers.get("location") ?? "")
```